lib/view/disclaimer_view.dart

```diff
@@ -1,206 +1,202 @@
       height: 44.0,
       child: RaisedButton(
         color: AppColors.green500,
         shape: const RoundedRectangleBorder(
           borderRadius: BorderRadius.all(Radius.circular(8)),
         ),
         onPressed: () {
+          // Write agreement of disclaimer to device storage to persist this decision
           _setAgreed();
+          // Make sure user cannot press back to return to the disclaimer once accepted
           Navigator.pushReplacementNamed(context, Routes.home);
         },
         child: const Text(
-          'I Agree',
-          style: AppStyles.textH5,
+          Strings.disclaimerButtonAgreeText,
+          style: Styles.textH5,
         ),
       ),
     );
 
-    return FutureBuilder<bool>(
-        future: _checkAgreed(),
-        builder: (context, snapshot) {
-          if (!snapshot.hasData) {
-            return const CircularProgressIndicator();
-          }
-          return WillPopScope(
-            // Prevent back button exiting disclaimer on Android
-            onWillPop: () async {
-              return _closeDisclaimerOrCloseApp(context, snapshot.data);
-            },
-            child: Scaffold(
-              backgroundColor: Colors.white,
-              appBar: AppBar(
-                backgroundColor: Colors.white,
-                elevation: 0.0,
-                iconTheme: AppStyles.appBarIconTheme,
-                automaticallyImplyLeading: snapshot.data ?? true,
-                title: Text(
-                  _title,
-                  style: AppStyles.textH5,
-                ),
-              ),
-              body: MediaQuery.of(context).size.height < 600
-                  ? Theme(
-                      data: ThemeData(accentColor: AppColors.green500),
-                      child: Stack(
-                        alignment: Alignment.bottomCenter,
-                        children: [
-                          NotificationListener<ScrollUpdateNotification>(
-                            onNotification: (scrollNotification) {
-                              _animationController.forward();
-                              return true;
-                            },
-                            child: ListView(
-                              children: <Widget>[
-                                _content,
-                                if (!snapshot.data)
-                                  _agreeButton
-                                else
-                                  _agreedText,
-                              ],
+    // Content of the page
+    Widget _agreedMessage(dynamic version, dynamic dateStampString) {
+      return Container(
+        padding: const EdgeInsets.all(16),
+        child: Text(
+          '${Strings.disclaimerHaveAgreedText} \nVersion: $version \nDate & time: $dateStampString',
+          style: Styles.textLegal,
+        ),
+      );
+    }
+
+    Widget _disclaimerScrollingContent(BuildContext context) {
```

Comment:
I would move this function outside the build method

Reply:
ditto 👍